packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -119,16 +132,26 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
     super.dispose();
   }
 
+  @override
+  void didUpdateConfig(CupertinoButton old) {
+    super.didUpdateConfig(old);
+    _opacityTween = new Tween<double>(
```

Comment:
i recommend factoring out the _opacityTween initialisation into a private method that's invoked from both initState and didUpdateConfig.